homeassistant/components/device_tracker/owntracks.py

```diff
@@ -60,47 +76,111 @@ def owntracks_event_update(topic, payload, qos):
             data = json.loads(payload)
         except ValueError:
             # If invalid JSON
-            logging.getLogger(__name__).error(
+            _LOGGER.error(
                 'Unable to parse payload as JSON: %s', payload)
             return
 
         if not isinstance(data, dict) or data.get('_type') != 'transition':
             return
 
-        # check if in "home" fence or other zone
-        location = ''
+        dev_id, host_name = _parse_topic(topic)
+
+        if data['desc'].lower() == 'home':
+            location = STATE_HOME
+        else:
+            location = data['desc']
+
+        kwargs = {
```

Comment:
Let's extract this into a method so that both times the kwargs are rendered the same. For example, you forgot the battery attribute here.